### PR TITLE
Merge built-in environments and discovered environemnts

### PR DESCRIPTION
**Pull request by dingmeng-xue**
Fix for this issue

https://github.com/Azure/azure-powershell/issues/12633

### CODE REVIEW

File: src/Authentication.Abstractions.Test/TestData/ArmResponseNoAzureCloud.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
+[
+  {
+    "portal": "https://portal.azure.us",
```
Comment (markcowl) [inline src/Authentication.Abstractions.Test/TestData/ArmResponseNoAzureCloud.json:3]: I think we might want to add a specific new piece of data with only one enbtry, just for completeness.  We want to make sure we protect ourselves from future bugs where we make assumptions about this payload.

File: src/Authentication.Abstractions.Test/AzureEnvironmentTests.cs [hunk 2]
```diff
@@ -52,6 +52,17 @@ public void TestArmCloudMetadata20190501Init()
...
         }
 
+        [Fact]
+        public void TestArmResponseNoAzureCloud()
+        {
+            Environment.SetEnvironmentVariable(ArmMetadataEnvVariable, @"TestData\ArmResponseNoAzureCloud.json");
+            var armEnvironments = AzureEnvironment.InitializeBuiltInEnvironments(null, httpOperations: TestOperationsFactory.Create().GetHttpOperations());
+
+            // Check AzureCloud is added to public environment list even discovery endpoint doesn't return AzureCloud.
+            Assert.True(AzureEnvironment.PublicEnvironments.ContainsKey(EnvironmentName.AzureCloud));
```
Comment (markcowl) [inline src/Authentication.Abstractions.Test/AzureEnvironmentTests.cs:62]: I would assert this for all of the built-in environments

File: src/Authentication.Abstractions.Test/TestData/ArmResponseNoAzureCloud.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
+[
+  {
```
Comment (markcowl) [inline src/Authentication.Abstractions.Test/TestData/ArmResponseNoAzureCloud.json:2]: I also got a report on the same thread that there are additional errors that prevent module load when customers use the  ```ARM_CLOUD_METADATA_URL``` environment variable to point at a metadata endpoint that does not have the 'AzureCloud' entry.  I believe we have a test using the environment variable for discovery, and we should ensure that it also convers this scenario.

Comment (dingmeng-xue) [reply src/Authentication.Abstractions.Test/TestData/ArmResponseNoAzureCloud.json:2]: You are right. The discovery approach via default DNS, parameter, or environment variable are going into the same logic. It should be covered.

File: src/Authentication.Abstractions.Test/AzureEnvironmentTests.cs (hunk 2 above)
Comment (dingmeng-xue) [reply src/Authentication.Abstractions.Test/AzureEnvironmentTests.cs:62]: Sure. I will add them.

File: src/Authentication.Abstractions.Test/TestData/ArmResponseNoAzureCloud.json (hunk 1 above)
Comment (dingmeng-xue) [reply src/Authentication.Abstractions.Test/TestData/ArmResponseNoAzureCloud.json:3]: Sure. I will add it.

**Comment by dingmeng-xue** [comment]
AzureEnvironment.InitializeBuiltInEnvironments is an internal method and public cloud map will be populated after this method. So, in the test case, check public environment variable is not correct because only InitializeBuiltInEnvironments is called. I only check public cloud environment is correctly returned in test cases.

**Pull request merged**